cmd/collector/main.go

```diff
@@ -70,11 +74,21 @@ func main() {
 	server := thrift.NewServer(ch)
 	server.Register(jc.NewTChanCollectorServer(jaegerBatchesHandler))
 	server.Register(zc.NewTChanZipkinCollectorServer(zipkinSpansHandler))
+
 	portStr := ":" + strconv.Itoa(*builder.CollectorPort)
 	listener, err := net.Listen("tcp", portStr)
 	if err != nil {
 		logger.Fatal("Unable to start listening on channel", zap.Error(err))
 	}
 	ch.Serve(listener)
-	select {}
+
+	r := mux.NewRouter()
```

Comment:
Have you considered using [yarpc](https://github.com/yarpc/yarpc-go) instead of mux? 
It looks like yarpc allows defining multiple inbounds (tchannel/http/etc) to the same thrift handler, and provides a unified way of adding interceptors, etc. 
I feel that it would cut down on some boiler plate, and having to do the conversion between http and tchannel ourselves.